posthog/views.py

```diff
@@ -241,3 +252,64 @@ def redis_values_view(request: HttpRequest):
     }
 
     return render(request, template_name="redis/values.html", context=context, status=200)
+
+
+@require_http_methods(["GET"])
+def preferences_page(request: HttpRequest, token: str) -> HttpResponse:
+    """Render the preferences page for a given recipient token"""
+    recipient, error = MessageRecipientPreference.validate_preferences_token(token)
+
+    if error:
+        return render(request, "message_preferences/error.html", {"error": error}, status=400)
+
+    if not recipient:
+        return render(request, "message_preferences/error.html", {"error": "Invalid recipient"}, status=400)
+
+    # Only fetch active categories and their preferences
+    categories = MessageCategory.objects.filter(deleted=False, team=recipient.team).order_by("name")
+    preferences = recipient.get_all_preferences() if recipient else {}
+
+    context = {
+        "recipient": recipient,
+        "categories": [
+            {
+                "id": cat.id,
+                "name": cat.name,
+                "description": cat.description,
+                "opted_in": preferences.get(cat.id) == PreferenceStatus.OPTED_IN,
+            }
+            for cat in categories
+        ],
+        "token": token,  # Need to pass this back for the update endpoint
+    }
+
+    return render(request, "message_preferences/preferences.html", context)
+
+
+@csrf_protect
+@require_http_methods(["POST"])
+def update_preferences(request: HttpRequest) -> JsonResponse:
```

Comment:
I'd still really wish there were some tests for this of some kind. Before release we really need some as this will be an important but very rarely tested flow...